### ISSUE TITLE
[Request] Endpoint for MarketNews Specialty Crops Custom Reports 

### ISSUE THREAD

**Issue by justinribeiro** [USDA/USDA-APIs#51]
Currently, a user can run custom reports for various specialty crops as part of the AMS (https://www.marketnews.usda.gov/mnp/fv-home).

At the moment, we're exporting XML by swapping the `commAbr` with a given commodity within the Agricultural Marketing Service reporting tool (an example being something like `https://www.marketnews.usda.gov/mnp/fv-report-top-filters?&commAbr=SWPOT&portal=fv&repType=shipPriceDaily&Go=Go&locName=All&type=shipPrice&navType=byComm&repDate=10%2F16%2F2017&endDate=10%2F16%2F2017&format=xml&rebuild=false`). 

This allows us to grab said XML by various means (curl, wget, internal software), but this feels fragile to us. If there any chance this system has an API endpoint that could be used instead?

**Comment by paulperrone** [comment]
I agree, this could use some work. Any devs on the project have relevant info to share?

**Comment by seanhtarget** [comment]
Also agree, looking for the same via api

**Comment by supersistence** [comment]
Would also love to see this happen. 
[Market Analysis & Reporting Services (MARS) API's](https://mymarketnews.ams.usda.gov/mars-api) [FAQs](https://mymarketnews.ams.usda.gov/mars-api/faqs) says that 
>All USDA Market News data will be available on the API in Spring 2019.

Spoke with [AMS](https://marketnews.usda.gov/mnp/fv-contactus) today and they confirmed a proposed March 2019 release date, but advised against holding one's breath. 

Would be interested to help make an R package for MARS API access. Anybody else?

**Comment by keithmichaelf** [comment]
> Would also love to see this happen.
> [Market Analysis & Reporting Services (MARS) API's](https://mymarketnews.ams.usda.gov/mars-api) [FAQs](https://mymarketnews.ams.usda.gov/mars-api/faqs) says that
> 
> > All USDA Market News data will be available on the API in Spring 2019.
> 
> Spoke with [AMS](https://marketnews.usda.gov/mnp/fv-contactus) today and they confirmed a proposed March 2019 release date, but advised against holding one's breath.
> 
> Would be interested to help make an R package for MARS API access. Anybody else?

I would be interested. This package is long overdue.

**Comment by keithmichaelf** [comment]
Is there any update on a MARS API for R?

**Comment by supersistence** [comment]
I haven't made any progress towards an API build.  
First step would be to see what, if anything, has been added to MARS.

**Comment by keithmichaelf** [comment]
> I haven't made any progress towards an API build.
> First step would be to see what, if anything, has been added to MARS.

Could we get started?

**Comment by osumo001** [comment]
is the api for the speciality crops still being built or completed?
